File: MusicEventManagementSystem/MusicEventManagementSystem.Client/src/frontend/event-organization/types/form/event.ts
import { EventStatus } from '../enums/EventOrganization';

export interface EventCreateForm {
  name: string;
  description: string;
  startDate: Date;
  endDate: Date;
  status: EventStatus;
  createdById: string;
  locationId: number;
}

export interface EventUpdateForm {
  name?: string;
  description?: string;
  startDate?: Date;
  endDate?: Date;
  status?: EventStatus;
  locationId?: number;
}

File: MusicEventManagementSystem/MusicEventManagementSystem.API/Models/Event.cs
using MusicEventManagementSystem.Enums;

namespace MusicEventManagementSystem.API.Models
{
    public class Event
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime Interval { get; set; }
        public EventStatus Status { get; set; } 
        public Guid CreatedById { get; set; }
        public int LocationId { get; set; }
        public Location Location { get; set; } = null!;
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
        
       // Navigation Property for One-to-One relationship with Negotiation
        public Negotiation? Negotiation { get; set; }
        // Navigation property - Event - (1,N) -> TicketType
        public ICollection<TicketType> TicketTypes { get; set; } = new List<TicketType>();

        // Navigation property - Event - (1,N) -> PricingRule
        public ICollection<PricingRule> PricingRules { get; set; } = new List<PricingRule>();

    }

}
